### ISSUE TITLE
Bug: ERROR program is unhealthy: lookup IP addresses do not match

### ISSUE THREAD

**Issue by tiberio87** [qdm12/ddns-updater#461]
**TLDR**: *Describe your issue in a one liner here*

1. Is this urgent:No
2. DNS provider(s) you use: Cloudflare
3. Program version: Latest

    

    `Running version latest built on 2020-03-13T01:30:06Z (commit d0f678c)`

4. What are you using to run the container: docker-compose
5. Extra information (optional)

Logs:

========================================
========================================
============= ddns-updater =============
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2023-04-17T11:30:42Z (commit 522cc7c)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-04-30T09:54:08+02:00 INFO reading JSON config from file /updater/data/config.json
2023-04-30T09:54:08+02:00 INFO Found single setting to update record
2023-04-30T09:54:08+02:00 INFO Reading history from database: domain thomeserver.biz host @
2023-04-30T09:54:08+02:00 INFO [healthcheck server] listening on 127.0.0.1:9999
2023-04-30T09:54:08+02:00 INFO [http server] listening on :8000
2023-04-30T09:54:08+02:00 INFO [backup] disabled
2023-04-30T09:54:09+02:00 INFO IPv4 address of ## is ## and your IPv4 address is ##
2023-04-30T09:54:09+02:00 INFO Updating record [domain: ## | host: @ | provider: cloudflare | ip: ipv4] to use ##
2023/04/30 09:54:43 "GET http://##:8001/ HTTP/1.1" from ##62516 - 200 1705B in 47.085µs

Configuration file (**remove your credentials!**):

```

{
    "settings": [
      {
        "provider": "cloudflare",
        "zone_identifier": "##",
        "domain": "##",
        "host": "@",
        "ttl": 600,
        "token": "##",
        "ip_version": "ipv4"
      }
    ]
  }

```

Host OS: Ubuntu

**Comment by tiberio87** [comment]
```log
Running version latest built on an unknown date (commit 5676673)

📣 Environment variables parsing was changed on 12 June, please report any issue you might have

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new

🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new

✨ New feature? https://github.com/qdm12/ddns-updater/issues/new

☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new

💻 Email? quentin.mcgaw@gmail.com

💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12

2023-06-12T16:57:50+02:00 INFO Settings summary:

├── HTTP client

|   └── Timeout: 10s

├── Update

|   ├── Period: 10m0s

|   └── Cooldown: 5m0s

├── Public IP fetching

|   ├── HTTP enabled: yes

|   ├── HTTP IP providers

|   |   └── all

|   ├── HTTP IPv4 providers

|   |   └── all

|   ├── HTTP IPv6 providers

|   |   └── all

|   ├── DNS enabled: yes

|   ├── DNS timeout: 3s

|   └── DNS providers

|       └── all

├── Resolver: use Go default resolver

├── IPv6

|   └── Mask bits: 128

├── Server

|   ├── Port: 8000

|   └── Root URL: /

├── Health

|   └── Server listening address: 127.0.0.1:9999

├── Paths

|   └── Data directory: /updater/data

├── Backup

|   ├── Period: 0s

|   └── Directory: /updater/data/backups

├── Logger

|   ├── Caller: no

|   └── Level: INFO

└── Shoutrrr

    ├── Addresses

    |   └── discord://*****

    └── Parameters

        └── title=DDNS Updater

2023-06-12T16:57:50+02:00 INFO reading JSON config from file /updater/data/config.json

2023-06-12T16:57:50+02:00 INFO Found 2 settings to update records

2023-06-12T16:57:50+02:00 INFO Reading history from database: domain ***.*** host @

2023-06-12T16:57:50+02:00 INFO Reading history from database: domain ***.***host www

2023-06-12T16:57:50+02:00 INFO [healthcheck server] listening on 127.0.0.1:9999

2023-06-12T16:57:50+02:00 INFO [http server] listening on :8000

2023-06-12T16:57:51+02:00 INFO [backup] disabled
```

**Comment by tiberio87** [comment]
With the latest update everything seems to work perfectly, thanks.

**Comment by qdm12** [comment]
@LeVraiRoiDHyrule Check the Cloudflare document again, you probably want to use the proxied field as true.

> The problem is that ddns-updater web ui is not hosted on mydomain.com but on ddns.mydomain.com, so the healthcheck server is also on that subdomain.

That doesn't matter / has influence on anything really.

**Comment by qdm12** [comment]
@goluftwaffe sorry for the delay answering;
```
IPv4 address of myhost.mydomain.com is newip and your IPv4 address is newip, skipping update
```

should mean there is no need to update, strange it's getting unhealhy. You can try the newer image 2.6 ot latest since many fixes were done during the last 3 weeks.

**Comment by goluftwaffe** [comment]
Thank you for the answer, I've move my dns name to cloudfare and all is working properly... I'm on the latest version.

**Comment by LeVraiRoiDHyrule** [comment]
> @LeVraiRoiDHyrule Check the Cloudflare document again, you probably want to use the proxied field as true.
> 
> > The problem is that ddns-updater web ui is not hosted on mydomain.com but on ddns.mydomain.com, so the healthcheck server is also on that subdomain.
> 
> That doesn't matter / has influence on anything really.

Thanks for your answer. I tried with the "proxied" field to true, for both ipv4 and ipv6, and I'm still having the same issue. DDNS updater is unable to check if the record has been updated and updates continuously.

**Comment by qdm12** [comment]
@LeVraiRoiDHyrule can you send your logs from the container running with `LOG_LEVEL=debug`? (make sure to remove credentials).

**Comment by LeVraiRoiDHyrule** [comment]
> @LeVraiRoiDHyrule can you send your logs from the container running with `LOG_LEVEL=debug`? (make sure to remove credentials).

Hi again. Sorry for the (very long) delay. I am still having this issue. About your advice to turn on the `proxied` setting in my config, this means turning on the "Proxy subdomain through cloudflare" setting at each update on Cloudflare admin panel, and I don't want to use this feature (I use Cloudflare as a simple DNS only). So I'll leave it to false.

With that said, here is my logs with `LOG_LEVEL=debug`:
```
2024-05-21T11:56:29+02:00 INFO reading JSON config from environment variable CONFIG
2024-05-21T11:56:29+02:00 DEBUG config read: {"settings": [{"provider": "cloudflare", "zone_identifier":"REDACTED", "domain":"REDACTED.com", "host":"@", "ttl": 1, "token":"REDACTED", "ip_version":"ipv4", "proxied":false},{"provider": "cloudflare", "zone_identifier":"REDACTED", "domain":"REDACTED.com", "host":"@", "ttl": 1, "token":"REDACTED", "ip_version":"ipv6", "proxied":false}]}
2024-05-21T11:56:29+02:00 INFO Found 2 settings to update records
2024-05-21T11:56:29+02:00 INFO Reading history from database: domain REDACTED.com host @ ipv4
2024-05-21T11:56:29+02:00 INFO Reading history from database: domain REDACTED.com host @ ipv6
2024-05-21T11:56:29+02:00 DEBUG configured to fetch IP: v4 or v6: false, v4: true, v6: true
2024-05-21T11:56:29+02:00 INFO [backup] disabled
2024-05-21T11:56:29+02:00 INFO [healthcheck server] listening on 127.0.0.1:9999
2024-05-21T11:56:29+02:00 INFO [http server] listening on :8000
2024-05-21T11:56:29+02:00 DEBUG your public IP address are: v4 or v6: invalid IP, v4: REDACTED, v6: REDACTED
2024-05-21T11:56:29+02:00 DEBUG record REDACTED.com (ipv4) is within cooldown period of 5m0s, skipping update
2024-05-21T11:56:29+02:00 DEBUG record REDACTED.com (ipv6) is within cooldown period of 5m0s, skipping update
```

The error of the health server is
```
ERROR program is unhealthy: lookup IP addresses do not match: 172.100.0.10,REDACTED instead of REDACTED for REDACTED.com 2024-05-21T12:01:29+02:00 INFO Shutdown successful 
```

To me, it looks like there is no real error in the container but the health server seems to be the only one in error.
Would you have an idea of what could be causing this ?
Thanks in advance and have a nice day.

**Comment by qdm12** [comment]
@LeVraiRoiDHyrule See https://github.com/qdm12/ddns-updater/discussions/726#discussioncomment-9786680
